pkg/infra/serverlock/serverlock.go

```diff
@@ -212,3 +252,16 @@ func (sl *ServerLockService) isLockWithinInterval(lock *serverLock, maxInterval
 	}
 	return false
 }
+
+func (sl ServerLockService) executeFunc(ctx context.Context, fn func(ctx context.Context)) {
+	start := time.Now()
+	ctx, span := sl.tracer.Start(ctx, "ServerLockService.executeFunc")
```

Comment:
Does it make sense to create a span in almost each method of this service?

Reply:
I think so, yes. A server lock isn't a cheap operation to start with, so it doesn't seem like a cardinality problem and it seems useful enough to be able to inspect how much time we're spending with each of the subcomponents.

![Screen Shot 2022-09-21 at 11 33 28](https://user-images.githubusercontent.com/298945/191470083-c180b4fe-6bd8-4dc3-b1e0-9d02c83ff28f.png)

Reply:
Thanks 👍

FYI If you enable database_metrics feature you'll get database driver traces as well

Reply:
Ah, I was wondering why I weren't getting those, thanks 🙂